compiler/compiler.go

```diff
@@ -162,6 +162,20 @@ func compile(n semantic.Node, typeSol semantic.TypeSolution, builtIns Scope, fun
 			object:   object,
 			property: n.Property,
 		}, nil
+	case *semantic.IndexExpression:
+		arr, err := compile(n.Array, typeSol, builtIns, funcExprs)
+		if err != nil {
+			return nil, err
+		}
+		idx, err := compile(n.Index, typeSol, builtIns, funcExprs)
+		if err != nil {
+			return nil, err
+		}
+		return &arrayEvaluator{
+			t:     monoType(typeSol.TypeOf(n)),
+			array: arr,
+			index: int(idx.EvalInt(builtIns)),
```

Comment:
This wont work when the index is an expression instead of a literal. The Eval call needs to happen as part of the Eval call of the arrayEvaluator.